lib/plugins/aws/deploy/compile/events/s3/index.js

```diff
@@ -13,9 +13,9 @@ class AwsCompileS3Events {
   }
 
   compileS3Events() {
-    if (!this.serverless.service.resources.Resources) {
-      throw new this.serverless.classes.Error(
-        'This plugin needs access to Resources section of the AWS CloudFormation template');
+    if (!this.serverless.service.provider.compiledCloudFormationTemplate) {
```

Comment:
Imho not necessary and should be covered by tests in createStack.js so that its always there